### PR TITLE
feat(cannon-contracts): MIPS64 conversion

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Converts `MIPS.sol` & `MIPS2.sol` to MIPS64.

### CODE REVIEW

**Comment by clabby** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11404?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#11404** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11404?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#11388** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/11388?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cannon/mipsevm/tests/evm_test.go [hunk 1]
```diff
@@ -386,7 +386,7 @@ func TestEVMFault(t *testing.T) {
 		nextPC uint64
 		insn   uint32
 	}{
-		{"illegal instruction", 0, 0xFF_FF_FF_FF},
```
Comment (clabby) [inline cannon/mipsevm/tests/evm_test.go:390]: This is a valid `SD` instruction, fyi

File: packages/contracts-bedrock/src/cannon/MIPS2.sol [hunk 2]
```diff
@@ -728,9 +730,9 @@ contract MIPS2 is ISemver {
         uint256 s = 0;
         assembly {
             s := calldatasize()
-            innerThreadRoot_ := calldataload(add(THREAD_PROOF_OFFSET, 166))
+            innerThreadRoot_ := calldataload(add(THREAD_PROOF_OFFSET, 322))
         }
         // verify we have enough calldata
-        require(s >= (THREAD_PROOF_OFFSET + 198), "insufficient calldata for thread witness"); // 166 + 32
+        require(s >= (THREAD_PROOF_OFFSET + 322), "insufficient calldata for thread witness"); // 356 + 32
```
Comment (benjaminion) [inline packages/contracts-bedrock/src/cannon/MIPS2.sol]: Something looks not right here - either `322` is wrong, or the comment is wrong, or (I suspect) both.

Comment (clabby) [reply packages/contracts-bedrock/src/cannon/MIPS2.sol]: Ah good spot; Yeah, both are wrong.

```suggestion
        require(s >= (THREAD_PROOF_OFFSET + 354), "insufficient calldata for thread witness"); // 322 + 32
```

**Comment by tynes** [comment]
Closing this due to https://github.com/ethereum-optimism/optimism/pull/12029 being merged

**Pull request closed without merging**